src/azure-cli/azure/cli/command_modules/resource/_params.py

```diff
@@ -228,6 +234,9 @@ def load_arguments(self, _):
         c.argument('handle_extended_json_format', arg_type=extended_json_format_type,
                    deprecate_info=c.deprecate(target='--handle-extended-json-format/-j'))
         c.argument('no_prompt', arg_type=no_prompt)
+        c.argument('confirm_with_what_if', arg_type=deployment_what_if_confirmation_type, is_preview=True)
+        c.argument('what_if_result_format', options_list=['--what-if-result-format', '-r'],
+                   arg_type=deployment_what_if_result_format_type, is_preview=True)
```

Comment:
Are these two parameters only related to deployment? Why are they also under the command group of resource?

Reply:
They are not. GitHub folds unchanged code so it looks like they are under the `resource` command group (notice that it's on line 66). If you expand you'll see that they are under `deployment create`.